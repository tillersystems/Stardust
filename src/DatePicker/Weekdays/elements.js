import styled from 'styled-components';

export const DayName = styled.div`
  display: flex;
  flex-basis: 2.8rem;
  justify-content: center;

  height: 1.1rem;
  width: 2.8rem;

  user-select: none;

  &:not(:nth-child(7n)) {
    margin-right: 0.8rem;
  }

  font-size: ${({
    theme: {
      fonts: { size },
    },
  }) => `calc(${size.default} - 0.2rem)`};
  text-transform: uppercase;

  color: ${({ theme: { palette } }) => palette.spaceGrey};
`;
