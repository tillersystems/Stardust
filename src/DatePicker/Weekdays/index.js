import React, { memo } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Info } from 'luxon';

import { DayName } from './elements';

/**
 * Week days
 *
 * This component is in charge of displaying
 * the week day names
 *
 * @param {string} className // Class needed by styled component.
 * @param {string} locale // Local to use to display the weekday names.
 *
 * @return {jsx}
 */

const Weekdays = ({ className, locale }) => {
  const dayNames = Info.weekdays('short', { locale });
  return (
    <div className={className}>
      {dayNames.map(dayName => (
        <DayName key={dayName}>{dayName}</DayName>
      ))}
    </div>
  );
};

/**
 * PropTypes Validation.
 */
const { string } = PropTypes;
Weekdays.propTypes = {
  className: string,
  locale: string,
};

/**
 * Default props.
 */
Weekdays.defaultProps = {
  className: '',
  locale: 'en',
};

export default memo(styled(Weekdays)`
  display: flex;
  flex-wrap: nowrap;
  margin: 1.6rem 0 0.8rem;
`);
