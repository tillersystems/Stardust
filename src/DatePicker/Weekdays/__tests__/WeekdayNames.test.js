import React from 'react';

import Weekdays from '..';

describe('<Weekdays />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Weekdays />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render with a different locale', () => {
    const { container } = render(<Weekdays locale="fr" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
