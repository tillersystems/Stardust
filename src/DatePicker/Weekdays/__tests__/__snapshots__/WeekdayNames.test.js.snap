// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weekdays /> should render with a different locale 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 2.8rem;
  -ms-flex-preferred-size: 2.8rem;
  flex-basis: 2.8rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1.1rem;
  width: 2.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: calc(1.2rem - 0.2rem);
  text-transform: uppercase;
  color: hsl(207,13%,45%);
}

.c1:not(:nth-child(7n)) {
  margin-right: 0.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin: 1.6rem 0 0.8rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Mon
  </div>
  <div
    class="c1"
  >
    Tue
  </div>
  <div
    class="c1"
  >
    Wed
  </div>
  <div
    class="c1"
  >
    Thu
  </div>
  <div
    class="c1"
  >
    Fri
  </div>
  <div
    class="c1"
  >
    Sat
  </div>
  <div
    class="c1"
  >
    Sun
  </div>
</div>
`;

exports[`<Weekdays /> should render without a problem 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 2.8rem;
  -ms-flex-preferred-size: 2.8rem;
  flex-basis: 2.8rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1.1rem;
  width: 2.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: calc(0.75rem - 0.2rem);
  text-transform: uppercase;
  color: hsl(207,13%,45%);
}

.c1:not(:nth-child(7n)) {
  margin-right: 0.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin: 1.6rem 0 0.8rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Mon
  </div>
  <div
    class="c1"
  >
    Tue
  </div>
  <div
    class="c1"
  >
    Wed
  </div>
  <div
    class="c1"
  >
    Thu
  </div>
  <div
    class="c1"
  >
    Fri
  </div>
  <div
    class="c1"
  >
    Sat
  </div>
  <div
    class="c1"
  >
    Sun
  </div>
</div>
`;
